src/qiskit_iqm/iqm_backend.py

```diff
@@ -74,3 +76,15 @@ def index_to_qubit_name(self, index: int) -> Optional[str]:
         """Given an index in the backend register return the corresponding IQM-style qubit name ('QB1', 'QB2', etc.).
         Returns None if the given index does not correspond to any qubit in the backend."""
         return self._idx_to_qb.get(index)
+
+    def validate_compatible_architecture(self, architecture: QuantumArchitectureSpecification) -> bool:
+        """Given a quantum architecture specification returns true if its number of qubits, names of qubits and qubit
+        connectivity matches the architecture of this backend."""
+        qubits_match = set(architecture.qubits) == set(self._quantum_architecture.qubits)
+        ops_match = set(architecture.operations) == set(self._quantum_architecture.operations)
+
+        self_connectivity = set(map(frozenset, self._quantum_architecture.qubit_connectivity))  # type: ignore
```

Comment:
mypy doesn't like `map(set, bla)`/`map(frozenset, bla)`, so I've added an ignore. The alternative is an imperative loop.